daemon/src/identify/dialer.rs

```diff
@@ -0,0 +1,158 @@
+use crate::identify::protocol;
+use crate::identify::PeerInfo;
+use crate::identify::PROTOCOL;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use tokio::sync::watch;
+use tokio_extras::spawn_fallible;
+use xtra::Address;
+use xtra::Context;
+use xtra_libp2p::endpoint;
+use xtra_libp2p::libp2p::PeerId;
+use xtra_libp2p::Endpoint;
+use xtra_libp2p::OpenSubstream;
+use xtra_productivity::xtra_productivity;
+
+pub struct Actor {
+    endpoint: Address<Endpoint>,
+    peer_infos: HashMap<PeerId, PeerInfo>,
+    peer_info_channel: Option<watch::Sender<Option<PeerInfo>>>,
```

Comment:
No longer that weird because I see that the maker doesn't care and passes a `None`. 

Perhaps it would be cleaner to have two constructors. Having to provide `None` is not very ergonomic.